src/models/converters/transactions/mod.rs

```diff
@@ -89,16 +89,16 @@ impl MultisigTransaction {
             .unwrap_or(false)
             && check_sender_or_receiver(&self.data_decoded, &self.safe)
         {
-            info_provider.token_info(&self.to).map_or(
-                TransactionInfo::Custom(self.to_custom()),
-                |token| match token.token_type {
+            match info_provider.token_info(&self.to) {
```

Comment:
While setting up mocks, I noticed that `map_or` used `Transaction::Custom(self.to_custom)` as the default **value**, so it was always evaluated. Even if it turned out to be a transfer. There is a version of `map_or_else` that uses a callback to evaluate the default if needed, but because `info_provider` is passed to 2 closures the compiler doesn't like it.

Long story short: I unfolded this stuff into a couple of match blocks to prevent the eager calling of `self.to_custom` which resulted into a nice little optimization. Maybe I could use `if let Ok(token) {  match token.type // ... } else { //.... }` if it's more readable.